src/charm.py

```diff
@@ -53,12 +66,25 @@ def register_event_handlers(self):
             self.s3_requirer.on.credentials_changed, self._on_s3_credential_changed
         )
         self.framework.observe(self.s3_requirer.on.credentials_gone, self._on_s3_credential_gone)
+        self.framework.observe(self.database.on.database_created, self._on_database_created)
+        self.framework.observe(self.database.on.endpoints_changed, self._on_database_created)
+        self.framework.observe(
+            self.on.database_relation_broken, self._on_database_relation_removed
+        )
         self.framework.observe(self.on.get_jdbc_endpoint_action, self._on_get_jdbc_endpoint)
 
     def _on_install(self, event: ops.InstallEvent) -> None:
         """Handle the `on_install` event."""
         self.unit.status = Status.WAITING_PEBBLE.value
 
+    def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
+        logger.info("Database created...")
+        self.update_service()
+
+    def _on_database_relation_removed(self, event) -> None:
+        logger.info("Database relation removed")
+        self.update_service()
```

Comment:
**todo** can we do a double check that this works as expected? both manually and with integration tests? The reason is that database removal events are somewhat curious, in that the relation still exists but it about to be torn down. What this means is that the data is still there and often in the logic we do checks on relation and/or data existence, thus not updating the service by removing the properties from the configuration files, until another `update_status` event is triggered.

This is actually the reason why in some removed events (in the new pattern, BTW at a certain point we should refactor the charm into the new pattern), we force the setting to "None" (see [here](https://github.com/canonical/spark-history-server-k8s-operator/blob/main/src/events/s3.py#L53)), to mimic that the relation was torn down (although the data is still there).

Reply:
During my tests, I've always found the `hive-site.xml` is updated by the relation broken event and never encountered the edge case you mentioned. I've also added a test case to check the removal of relation in the integration tests.